rfcs/0133-ipfs.md

```diff
@@ -0,0 +1,284 @@
+---
+feature: ipfs
+start-date: (fill me in with today's date, YYYY-MM-DD)
+author: John Ericsion (@Ericson2314) on behalf of [Obsidian Systems](https://obsidian.systems)
+co-authors: (find a buddy later to help out with the RFC)
+shepherd-team: (names, to be nominated and accepted by RFC steering committee)
+shepherd-leader: (name to be appointed by RFC steering committee)
+related-issues: (will contain links to implementation PRs)
+---
+
+# Summary
+[summary]: #summary
+
+Integrate Nix with IPFS, in phases of increasing sophistication.
+This follows the work done and described in https://github.com/obsidiansystems/ipfs-nix-guide/ .
+
+# Motivation
+[motivation]: #motivation
+
+## Binary distribution
+
+Currently distributing Nix binaries takes a lot of bandwidth and storage.
+This is a barrier to being a Nix user in areas of slower internet --- which includes the vast majority of the world's population at this time.
+This is also a barrier to users running their own caches.
```

Comment:
How is it addressed in Obsidian that IPFS (by default, I think) duplicates the storage requirement of a package? Both in the output directory `/nix/store/...` and in IPFS own blocks store.

Reply:
This is not yet addressed. The focus of this RFC is interfaces not implementation details. I think it is fine that is out of scope for now because we can transparently improve it later.

Also Obsidian Systems is just the name of the company I work at. This project doesn't have a "branded" name, because it is meant to become a seamless part of upstream Nix.

Reply:
SGTM